### PR TITLE
Devs

### PR DESCRIPTION
**Pull request by Dzionek95**
Hello !

There are a lot of changes in controller connected with integration of gui with client and controller

### CODE REVIEW

File: src/main/java/com/java_academy/ServerApp.java [hunk 1]
```diff
@@ -24,25 +26,27 @@ public static void main(String[] args) {
         try {
             ServerSocket serverSocket = new ServerSocket();
             SocketProvider socketProvider = new ServerSocketProvider(serverSocket);
-
             InetSocketAddress inetSocketAddress = new InetSocketAddress("localhost", 3000);
-
             Connector connector = new Connector(socketProvider);
 
-            connector.addMessageReceiverListenerToSocketProvider(messageSupplier -> {
+/*            connector.addMessageReseiverListenerToSocketProvider(messageSupplier -> {
```
Comment (ghost) [inline src/main/java/com/java_academy/ServerApp.java:32]: :-1:

File: src/main/java/com/java_academy/logic/state_machine/GetBoardForPlayer.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        displayConsumer.accept(new MessageObject(Players.FIRST_PLAYER, MessageCreator.createJsonMarkedIndexes(firstPlayerBoard)));
+        displayConsumer.accept(new MessageObject(Players.SECOND_PLAYER, MessageCreator.createJsonMarkedIndexes(secondPlayerBoard)));
+    }
+
+    @Override
+    public GameState changeState(String message) {
+        System.out.println("Get Board for player");
+        return new PlayerActionState(Players.FIRST_PLAYER);
+    }
+}
```
Comment (ghost) [inline src/main/java/com/java_academy/logic/state_machine/GetBoardForPlayer.java:36]: :+1:

File: src/main/java/com/java_academy/gui/Controller.java [hunk 3]
```diff
@@ -1,37 +1,74 @@
...
 
+import java.net.InetSocketAddress;
+import java.net.Socket;
 import java.net.URL;
-import java.util.List;
+import java.util.Map;
 import java.util.ResourceBundle;
+import java.util.function.Supplier;
 
 public class Controller implements Initializable {
```
Comment (ghost) [inline src/main/java/com/java_academy/gui/Controller.java:30]: To refactor, but it's ok. I know that it's prototype

**Review by ghost — APPROVED** [review]

File: src/main/java/com/java_academy/ClientApp.java [hunk 4]
```diff
@@ -31,17 +31,15 @@ public static void main(String[] args) {
...
             System.out.println("Sending message \"" + message + "\" to the server");
             connector.sendMessage(new MessageObject(Players.FIRST_PLAYER, message));
 
-            Connector.getExecutor().schedule(new Runnable() {
-                @Override
-                public void run() {
-                    connector.sendMessage(new MessageObject(Players.FIRST_PLAYER, CLOSE_MESSAGE));
-                    connector.closeConnection();
-                }
+            Connector.getExecutor().schedule(() -> {
```
Comment (mhbratek) [inline src/main/java/com/java_academy/ClientApp.java:40]: Nice :1st_place_medal:

File: src/main/java/com/java_academy/ClientApp.java [hunk 5]
```diff
@@ -31,17 +31,15 @@ public static void main(String[] args) {
...
+                connector.sendMessage(new MessageObject(Players.FIRST_PLAYER, CLOSE_MESSAGE));
+                connector.closeConnection();
             }, 4, TimeUnit.SECONDS);
```
Comment (mhbratek) [inline src/main/java/com/java_academy/ClientApp.java:43]: 4 means ? Magic number.

File: src/main/java/com/java_academy/ServerApp.java (hunk 1 above)
Comment (mhbratek) [reply src/main/java/com/java_academy/ServerApp.java:32]: :-1:

File: src/main/java/com/java_academy/logic/state_machine/GetBoardForPlayer.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    private MarkedIndexes secondPlayerBoard;
+
+    public GetBoardForPlayer(MarkedIndexes firstPlayerBoard, MarkedIndexes secondPlayerBoard) {
+		this.firstPlayerBoard = firstPlayerBoard;
+		this.secondPlayerBoard = secondPlayerBoard;
+	}
+
+	@Override
+    public void display(Consumer<MessageObject> displayConsumer) {
+        displayConsumer.accept(new MessageObject(Players.FIRST_PLAYER, MessageCreator.createJsonMarkedIndexes(firstPlayerBoard)));
```
Comment (mhbratek) [inline src/main/java/com/java_academy/logic/state_machine/GetBoardForPlayer.java:27]: Maybe divide this lines, because they are too long.

**Review by mhbratek — APPROVED** [review]
Good Job.

**Pull request merged**